### PR TITLE
v3 release

### PR DESCRIPTION
**Pull request by cole-brokamp**
tracking progress in #6

### CODE REVIEW

**Comment by erikarasnick** [comment]
@cole-brokamp I'm having trouble building the container. I get all the way to the 'make install' step, then get this error 

`cp: cannot create regular file '../lib/geocoder/us/sqlite3.so': No such file or directory`

which I have tracked to the Makefile in the src directory. Not sure if this is something where you can relatively easily tell what is going wrong, or if it would be useful to interactively troubleshoot. 

I still have work to do on this (switching to the newer ruby file with standard outputs, changing the geocode function in the R script to be more like the batch geocoder, updating readme, etc) but thought it would be useful to get this working first.

**Comment by cole-brokamp** [comment]
I think this should get you to having the geocoder built and just having to do the R and ruby file stuff.  I commented those parts out just for the purposes of troubleshooting the container build so far.

**Comment by erikarasnick** [comment]
Everything else seems to build fine, but when I try to test, I get this error: 

`docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "geocode.R": executable file not found in $PATH: unknown.`

When I shell into the container the file is there.

**Comment by cole-brokamp** [comment]
Can you push up the code?  How are you calling "geocode.R"?  Can you call it when you shell in?

**Comment by erikarasnick** [comment]
I haven't changed much. I tried with "#!/usr/local/bin/Rscript" and "#!/usr/bin/Rscript".

File: Dockerfile [hunk 1]
```diff
@@ -32,23 +33,21 @@ RUN cd /root/geocoder \
...
-# COPY renv.lock .
-# RUN R --quiet -e "renv::restore(repos = c(CRAN = 'https://packagemanager.rstudio.com/all/__linux__/focal/latest'))"
+COPY geocode.R .
+COPY geocode.rb .
 
-# COPY geocode.R .
-# COPY geocode.rb .
-
-# ENTRYPOINT ["geocode.R"]
+ENTRYPOINT ["geocode.R"]
```
Comment (cole-brokamp) [inline Dockerfile:53]: I think this has to be `./geocode.R` otherwise its looking for an executable on $PATH that is called "geocode.R".

**Comment by erikarasnick** [comment]
changed to 
`ENTRYPOINT ["./geocode.R"]`

and got
`standard_init_linux.go:219: exec user process caused: no such file or directory`

**Comment by cole-brokamp** [comment]
what about when you shell into the container?  can you call `./geocode.R` there?  it might be not starting in the right working directory.  maybe use `./root/geocoder/geocode.R` instead?

**Comment by erikarasnick** [comment]
When I shell in I am automatically in the `/geocoder` directory, which is where `geocode.R` is. 

When I call `./geocode.R` I get `bash: ./geocode.R: /usr/local/bin/Rscript: bad interpreter: No such file or directory`

I tried `./root/geocoder/geocode.R` yesterday and just tried again. Both times:
`docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "./root/geocoder/geocode.R": stat ./root/geocoder/geocode.R: no such file or directory: unknown.`

**Comment by cole-brokamp** [comment]
OK.  I think the path to Rscript is wrong in the R file?  When you shell in, what does `which Rscript` return?

**Comment by erikarasnick** [comment]
`/usr/bin/Rscript`

**Comment by cole-brokamp** [comment]
Change the shebang in `geocode.R` to that path and it should work.

**Comment by erikarasnick** [comment]
I changed it, removed all cached images and built from scratch, but I still get the same error 

`docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "geocode.R": executable file not found in $PATH: unknown.`

**Comment by cole-brokamp** [comment]
Still need to use `./geocode.R` in the entrypoint

**Comment by erikarasnick** [comment]
gotcha! it's working now.

**Comment by cole-brokamp** [comment]
Can we change it around so that a "make build" from the console triggers the Docker build and not the makefile used for the geocoder install?

**Comment by erikarasnick** [comment]
I had tried it that way and was having issues, but now that everything else is working I switched the makefile names again.

I was also able to use `ADD`. Not sure what was going on before.

**Comment by erikarasnick** [comment]
yep, done.

**Comment by erikarasnick** [comment]
I included `digest` because of this error 

Failed with error:  'there is no package called 'digest''
Error in memoise::cache_filesystem(cache.name) :
  Package `digest` must be installed for `cache_filesystem()`.

which made me think it had to do with `mappp`?

I will look into using kable. Also, I've been noticing weird characters in the cli messages printed to the console. Did you see anything like this:

`i<U+00A0>geocoding complete; now filtering to precise geocodes...`

**Comment by cole-brokamp** [comment]
Yes, looks like `digest` is suggested, but not required: https://github.com/r-lib/memoise/blob/bc88bcad58c2ba4223711a598ec21071f88a4532/DESCRIPTION#L39  But this is required for our purposes.  But can we eliminate its startup message?

I'm also seeing weird unicode characters from the cli package.  I thought it was because I was using emacs, but that's definitely a problem if its showing up in rstudio and terminal too

**Comment by erikarasnick** [comment]
just realized I need to fix the cli messages in the dht package, though.

**Pull request merged**